tasks/main.yml

```diff
@@ -1,5 +1,9 @@
 ---
-- include_tasks: prerequisite.yml
+- include_tasks: ../roles/hp-firmware-upgrade/tasks/find_hp_gen.yml
+  tags:
+    - find_hp_gen
+
+- include_tasks: ../roles/hp-firmware-upgrade/tasks/prerequisite.yml
```

Comment:
I don't understand why we would want to change the path of the task? Maybe this is some practice that I'm not aware of.

Reply:
The problem is, that this role use a main.yml as starting point. If you use a playbook to start the tasks in the role and you include the role, the sequence of the tasks in the role are alphabetical. So I insert the include task in the playbook so we can start with the main.yml. The problem is, that the include_task directive in the role uses the playbook directory, too. So we have to set the relative path in the role task to successfully execute each task in the role.

Reply:
Hi, thats realy strange. I will reproduce this behavior as soon as possible. 
Here is my playbook thats actually running:

`- name: Upgrade HP Firmware
  hosts: all
  vars:
    upgrade_ilo_firmware: False
    upgrade_system_rom_firmware: False
    upgrade_disk_controller_firmware: False
    upgrade_disk_drive_firmware: False
    upgrade_power_management_controller_firmware: False
    upgrade_intel_network_adapter_firmware: False
    upgrade_mellanox_network_adapter_firmware: False
    upgrade_qlogic_network_adapter_firmware: False
  tasks:
    - include_tasks: ../roles/hp-firmware-upgrade/tasks/main.yml`

Reply:
Thanks for checking. 
Always fun to hear about new uses of ansible! I also edited your message to make it look nice in github :)